automerge/src/query.rs

```diff
@@ -185,6 +185,108 @@ impl Default for Index {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Default)]
+pub(crate) struct VisWindow {
+    counters: HashMap<OpId, CounterData>,
+}
+
+impl VisWindow {
+    fn visible(&mut self, op: &Op, pos: usize) -> bool {
```

Comment:
Can we make the clock for `visible_at` optio Al and then just pass `None` instead of nearly duplicating the implementation?

Reply:
I could - I guess I was trying to not add any extra "if" statements to the fast path.  But now that I think about it one of the goals for this codebase I have is not adding any optimizations to the code that add complexity if we can't show a measurable difference in the benchmarks.  I suspect that extra "if" wont move the needle at all.

Reply:
Well I tried and combining the two functions into one cost me about 20ms on the benchmark (375ms -> 395ms).  Alas...

Reply:
Ah well, at least we know it's worth it.